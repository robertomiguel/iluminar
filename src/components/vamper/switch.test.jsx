import { fireEvent, render, screen } from '@testing-library/react'
import { Switch } from './switch'

describe('Prueba del Switch', () => {
    test('match con snapshot', () => {
        const { container } = render(<Switch />)
        expect(container).toMatchSnapshot()
    })

    test('si está activo muestra SI', () => {
        const { container, getByText } = render(<Switch checked={true} />)
        const button = getByText('SI')
        expect(button).toBeTruthy()
    })

    test('si no está activo muestra NO', () => {
        const { container, getByText } = render(<Switch checked={false} />)
        const button = getByText('NO')
        expect(button).toBeTruthy()
    })

    test('screen test cuando checked es false', () => {
        render(<Switch checked={false} />)
        expect(screen.getByText('NO')).toBeTruthy()
        // el screen toma el Virtual DOM
    })

    test('onClick: Pasar de NO a SI', () =>{
        render(<Switch checked={false} />)
        fireEvent.click(screen.getByText('NO'))
        expect(screen.getByText('SI')).toBeTruthy()
        fireEvent.click(screen.getByLabelText('sw-button'))
        expect(screen.getByText('NO')).toBeTruthy()
    })
})