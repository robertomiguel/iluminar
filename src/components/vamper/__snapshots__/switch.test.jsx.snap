// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba del Switch match con snapshot 1`] = `
<div>
  <div
    style="text-align: center; display: flex; flex-direction: column; align-content: center; justify-content: center; align-items: center; gap: 1px;"
  >
    <div
      style="color: white; font-size: 16px; letter-spacing: 2px; background: black; padding: 5px; border-radius: 7px;"
    />
    <div
      style="display: flex; background: black; color: gray; border: 1px solid yellow; border-radius: 15px; overflow: hidden; width: 80px;"
    >
      <div
        aria-label="sw-button"
        style="display: flex; justify-content: center; align-items: center; width: 80px; height: 30px; cursor: pointer;"
      >
        NO
      </div>
    </div>
  </div>
</div>
`;
